src/core.ts

```diff
@@ -126,63 +126,18 @@ export const withEnvironment = curry(<M>(env: Environment, def: ContainerDef<M>)
- *         - `update`: A function that accepts the current model as an object,
- *            and returns a `Map` pairing message types to handler functions.
- *            Each handler function accepts the message's data as a parameter,
- *            returns an update result. See "Update Results" below for details.
- *         - `view`: A pure component (i.e. a function that returns React DOM) — as part of
- *           its props, the component will receive an `emit` function, which can be called
- *           with a user-defined message class.
- *         - `attach`: Optional. If attaching to a Redux store, pass `store` and (optionally) `key`. The
- *           container will then subscribe to the store and propagate changes. Remember to call `reducer()`
- *           on the container and pass the resulting value to `combineReducers()` to complete the loop.
- *
- * ### Update Results
- *
- * An update result is a value returned from dispatching a message. Update results are used
- * to update the containers's model, as well as return _command messages_, which represent
- * side-effects that the container can perform, such as an HTTP request.
- *
- * Update results can be one of the following:
- *
- * - An object: to update the model without performing any actions — at a minimum, the
- *   current model must always be returned
- * - An array of `[model, message]`: to update the model _and_ perform an action, return an array
- *   with the model first, and a message object second,
- *   i.e. `[model, new Alert({ message: "Hello world!" })]`
- * - An array of `[model, [message]]`: to perform multiple actions, simply return an array of
- *   command messages
- *
- * @return {Object} returns an object with the following methods:
- *
- *  - `dispatch`: Accepts a message object to update the container's model
- *  - `state`: Returns the current model
- *  - `view`: A React wrapper component that can be rendered or embedded
- *    in another component
- *  - `reducer`: Returns a Redux-compatible reducer function. Optionally accepts a hash
- *    pairing action types to message types, to enable the container to respond to Redux actions,
- *    which will be mapped to messages.
- *  - `push`: Accepts a new model value to update the container.
- *  - `subscribe`: Accepts a callback which receives a copy of the model when it is updated.
- *  - `identity`: Returns an object containing the original values that created this container.
- *  - `accepts`: Accepts a message class and returns a boolean indicating whether the container
- *    accepts messages of that type.
- */
-export const container: <M>(def: ContainerDef<M>) => Container<M> = withEnvironment(root);
-
 /**
  * Returns a copy of a container, disconnected from its effects / command dispatcher.
  * Calling `dispatch()` on the container will simply return any commands issued.
  */
 export const isolate = <M>(ctr: Container<M>, opts: any = {}): IsolatedContainer<M> => {
```

Comment:
Shouldn't a type be used other than `any`? Seems pretty bad practice to continue developing agains a non-typed argument especially when TypeScript, and this "framework" have been around for years.

Reply:
> Shouldn't a type be used other than `any`?

Probably so! The API was in flux for a while, and I don't think we ever went back and refined it. Plus, it's a testing method, so, pretty low-stakes.

> Seems pretty bad practice to continue developing agains a non-typed argument especially when TypeScript, and this "framework" have been around for years.

Well, the nice thing about Open Source is that no one forces anyone to use it, so if you think a project uses bad practices, you can just avoid it and move on with your life. 😉